tests/unit/resolve-repositories.test.js

```diff
@@ -0,0 +1,188 @@
+import { resolveRepositories } from '../../lib/resolve-repositories.js';
+import { Octokit } from "@octoherd/octokit";
+import nock from "nock";
+import { suite } from "uvu";
+import { equal } from "uvu/assert";
+
+const resolveRepos = suite('resolve-repositories')
+const withOrg = suite('when GitHub account is an Organization: "@octoherd"')
+const withUser = suite('when GitHub account is a User: "@gr2m"')
+
+withOrg('when single repository exists', async () => {
+	const org = 'octoherd';
+	const repo = 'cli';
+	const octokit = new Octokit({
+		auth: 'randomToken',
+	});
+	
+	const repositories = [
+		`${org}/${repo}`
+	]
+
+	const mockedResponse = { name: repo }
+	
+	nock('https://api.github.com')
+		.get(`/repos/${org}/${repo}`)
+		.reply(200, mockedResponse)
```

Comment:
In unit tests, there should be no need for mocking http requests, that is something we would do in integration tests.

Instead I would create a mock `octokit` object that doesn't send any request. You can do something like this

```js
await resolveRepositories({ log: console, octokit: { request: () => {}, paginate: { iterator: () => {} }, repositories)
```

and replace the `() => {}` with mock methods. You can create the mock functions and assertions how they have been called yourself or use a simple standalone mock library such as https://github.com/jupiter/simple-mock, if you prefer. I've used `simple-mock` a lot in the past, maybe there is a more modern alternative, I haven't looked for a while

Reply:
I researched a bit on GitHub projects using `uvu` and I did not find a clear winner as a mock solution so I ended up using `simple-mock`.

What I found is [this issue](https://github.com/lukeed/uvu/issues/10) on `uvu`'s repo, where they suggest [`snoop`](https://github.com/therealparmesh/snoop) and [`sinon`](https://github.com/sinonjs/). 

I think sinon is [too heavy](https://bundlephobia.com/result?p=sinon@10.0.0) for the purpose of being fast and light and [snoop](https://www.npmjs.com/package/snoop) has a similar weight and adopton than [`simple-mock`](url). My only concern is the unmaintainability of simple-mock:

![image](https://user-images.githubusercontent.com/2574275/115119494-ea59ae00-9fa8-11eb-8724-ecbe4a67a092.png)

Source: https://www.npmtrends.com/simple-mock-vs-sinon-vs-snoop

💡 If you want we can experiment with other options in a different PR and run benchmarks. @gr2m 

---

I found @kentcdodds used uvu so I asked him. Let's see if he has any suggestions for mocking.

Reply:
I only used uvu because jest wasn't quite ready for my needs on mdx-bundler and the native ESM modules it uses. I still recommend jest when you can use it. It's mocking capabilities are great.

Not sure why you're concerned about the size for a testing utility 🤷  But yeah, I don't have any more useful advise here. Sorry.

Reply:
> I only used uvu because jest wasn't quite ready for my needs on mdx-bundler and the native ESM modules it uses. I still recommend jest when you can use it. It's mocking capabilities are great.

Thanks for jumping here and for your time and feedback on this topic. 

> Not sure why you're concerned about the size for a testing utility 🤷 But yeah, I don't have any more useful advise here. Sorry.

I guess the size can be a parameters to consider if it increases the execution time of a Github Action running a `test` check run for exemple (dowloading the dependency or adding overhead at runtime).